### ISSUE TITLE
System.InvalidOperationException: Headers are read-only

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#1950]
### What happened?

Error messages like this repeatedly show in the website log file

2023-10-23 23:00:35.4243 | 17|ERROR| Connection id "0HMUJTKD93BNV", Request id "0HMUJTKD93BNV:00000001": An unhandled exception was thrown by the application. System.InvalidOperationException: Headers are read-only, response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpHeaders.ThrowHeadersReadOnlyException()
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseHeaders.Microsoft.AspNetCore.Http.IHeaderDictionary.set_ContentType(StringValues value)
   at Microsoft.AspNetCore.Http.DefaultHttpResponse.set_ContentType(String value)
   at IsraelHiking.API.Services.Middleware.SpaDefaultHtmlMiddleware.InvokeAsync(HttpContext context) in /net/IsraelHiking.API/Services/Middleware/SpaDefaultHtmlMiddleware.cs:line 38
   at IsraelHiking.API.Services.Middleware.CrawlersMiddleware.InvokeAsync(HttpContext context, IDetectionService detectionService) in /net/IsraelHiking.API/Services/Middleware/CrawlersMiddleware.cs:line 102
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)

### What steps will reproduce the bug?

Open the log file

### What I expect to happen

No such error in the log file

### Platform

- [ ] Israel Hiking Map app
- [ ] Israel Hiking Map site in a browser

### OS Name and Version

Server side

### Browser Name and Version

Server side

**Comment by HarelM** [comment]
Related to the following part of the code I believe:
https://github.com/IsraelHikingMap/Site/blob/645b0544889446f35de97a0fb30a89c60ec9c4aa/IsraelHiking.API/Services/Middleware/SpaDefaultHtmlMiddleware.cs#L38
I need to figure out why this is incorrect...

**Comment by HarelM** [comment]
The problem as can be seen in the logs is that we return a response from the CrawlerAPIMiddleware and then try to update this response from the SpaMiddleware, which is incorrect.
I have updated the code and the tests to avoid this issue.
I have tested to see that this has no impact on the server besides the logs and made sure I understand the root cause of the issue, next version should not have these errors.
